web/app/js/components/TabbedMetricsTable.jsx

```diff
@@ -92,7 +94,76 @@ const nameToDataKey = {
   successRate: "SUCCESS_RATE",
   latency: "LATENCY"
 };
+
 export default class TabbedMetricsTable extends React.Component {
+  constructor(props) {
+    super(props);
+    this.api = ApiHelpers(this.props.pathPrefix);
+    this.handleApiError = this.handleApiError.bind(this);
+    this.loadFromServer = this.loadFromServer.bind(this);
+
+    let tsHelper = urlsForResource(this.props.pathPrefix, this.props.metricsWindow)[this.props.resource];
+
+    this.state = {
+      timeseries: {},
+      groupBy: tsHelper.groupBy,
+      metricsUrl: tsHelper.url(this.props.entity),
+      error: '',
+      lastUpdated: this.props.lastUpdated,
+      metricsWindow: "10s",
+      pollingInterval: "10000",
+      pendingRequests: false
+    };
+  }
+
+  componentDidMount() {
+    if (!this.props.hideSparklines) {
+      this.loadFromServer();
+      this.timerId = window.setInterval(this.loadFromServer, this.state.pollingInterval);
```

Comment:
realizing we have this pattern in a lot of places, though would still like your thoughts on setting a timer upon completion of `loadFromServer`. this would prevent requests from piling up if `loadFromServer` takes nearly as long as `pollingInterval`.

Reply:
this method of polling is definitely still rough :) do you mean so the server isn't continuously working if loadFromServer takes a long time? we do have the `pendingRequests` check such that we won't issue a duplicate request for the same thing if we already have issued a request, so requests should at least not be accumulating there.